### ISSUE TITLE
Invalid result for BC::pow with negative fractional exponent

### ISSUE THREAD

**Issue by TheLevti** [krowinski/bcmath-extended#33]
> Please provide the following details.

* *Operating System*: Linux
* *PHP Version*: 7.2

> Steps required to reproduce the problem.

For testing I used another library https://php-decimal.io. Following code compares the result of `BC::pow` and `$decimal->pow`

```php 
public function testNegativeExponentBCVsDecimal()
{
        $base = '1.005';
        $exp = '-120.12345678';
​
        $v1Dec = new Decimal($base);
        $v1 = $v1Dec->pow(new Decimal($exp))->toString();    
        $v1_old = BC::pow($base, $exp);    
        printf("\n$v1\n$v1_old");
}
```

> Expected Result.

`0.5492944034820568190269179210`

> Actual Result.

`0.5499712716356165`

So the BC version will start deviating after the third digit after the comma. It does not matter how high you set the scale, it will always be incorrect after a few digits.

**Comment by TheLevti** [comment]
I checked with wolfram alpha and other tools and `0.5492944034820568190269179210...` should be the correct answer.

**Comment by TheLevti** [comment]
I think tests did not cover this case unfortunately.

**Comment by peter279k** [comment]
Hi @TheLevti, thanks for your report.

I also have the same issue from sample code you provide.

It looks like the `php-decimal` is the customized [PHP extension](https://github.com/php-decimal/ext-decimal).

And it looks like this issue is about the `BC` math extension issue.

I also recommend submitting this bug to official[ PHP bug report page](https://bugs.php.net).

**Comment by krowinski** [comment]
The problem with bcpow is that second parameter can be only integer 

```
exponent
The exponent, as a string. If the exponent is non-integral, it is truncated. The valid range of the exponent is platform specific, but is at least -2147483648 to 2147483647.
```

so I try to calculate using BC::powFractional and it's not as accurate as its should in all cases I will try to find solution in free time.

Good to know that someone made php-decimal.

**Comment by peter279k** [comment]
@krowinski, just notice that the `php-decimal` is PHP extension wrapper and a `libmpdec` binding.

**Comment by krowinski** [comment]
fix in 5.1 tag
